schema/proto.go

```diff
@@ -264,6 +254,11 @@ func generateSchemaTimestamp(item *Item) (*parquet.SchemaElement, *common.Tag, e
 	return schema, newInfo, nil
 }
 
+// DFS traverse the obj underlying struct type. If field is strucd recursive visit its sub-fields.
```

Comment:
I noticed you're explicitly using a `stack` in your DFS as opposed to leveraging **recursion**. Is this due to some performance concerns or ending up with **too deep call-stacks** since a Struct can be heavily nested at times?

Reply:
Yes. Performance is one thing. In addition, they have similar implementation for other flow.